### PR TITLE
Implement LWG-3648: `format` should not print `bool` with `'c'`

### LINKED ISSUE #2555: LWG-3648 `format` should not print `bool` with `'c'`
LWG-3648 `format` should not print `bool` with `'c'`

### PR DESCRIPTION
**Pull request by fsb4000**
Fixes #2555

### CODE REVIEW

**Comment by cpplearner** [comment]
`test_sane_c_specifier` should be renamed as there's no "sane" `'c'` specifier now.

**Review by StephanTLavavej — APPROVED** [review]

File: tests/libcxx/expected_results.txt [hunk 1]
```diff
@@ -31,6 +31,12 @@ std/language.support/support.exception/propagation/rethrow_exception.pass.cpp FA
 # Testing nonstandard behavior
 std/utilities/template.bitset/bitset.cons/string_ctor.pass.cpp FAIL
 
+# libc++ doesn't implement LWG-3648: "format should not print bool with 'c'"
```
Comment (CaseyCarter) [inline tests/libcxx/expected_results.txt:34]: It's weird to add new skips in the middle of a section; could we please move this down to line 88-ish? (And in `skipped_tests.txt` as well).

**Review by CaseyCarter — CHANGES_REQUESTED** [review]

File: tests/libcxx/expected_results.txt (hunk 1 above)
Comment (CaseyCarter) [reply tests/libcxx/expected_results.txt:34]: (This is simple and mechanical, I'll just apply the change.)

**Review by CaseyCarter — APPROVED** [review]

**Review by barcharcraz — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for implementing this LWG issue resolution that prevents this wacky combination! :zany_face: :smiley_cat: :tada:

**Pull request merged**